packages/solid/src/reactive/signal.ts

```diff
@@ -1185,8 +1185,9 @@ export function createContext<T>(
  * @description https://docs.solidjs.com/reference/component-apis/use-context
  */
 export function useContext<T>(context: Context<T>): T {
-  return Owner && Owner.context && Owner.context[context.id] !== undefined
-    ? Owner.context[context.id]
+  let value: undefined | T;
+  return Owner && Owner.context && (value = Owner.context[context.id]) !== undefined
```

Comment:
how about
```tsx
Owner?.context?.[context.id] ?? context.defaultValue
```

Reply:
I've been avoiding optional chaining as we don't transform JS in the repo and some platforms down't support optional chaining. We've had issue trying to introduce it in the past.